tensorflow/compiler/xla/service/gpu/cusolver_context.cc

```diff
@@ -17,6 +17,64 @@ limitations under the License.
 
 #include "tensorflow/compiler/xla/util.h"
 
+namespace rocblas_wrap {
+
+using stream_executor::internal::CachedDsoLoader::GetRocblasDsoHandle;
+using tensorflow::Env;
+
+#ifdef PLATFORM_GOOGLE
+#define ROCBLAS_API_WRAPPER(__name)           \
+  struct WrapperShim__##__name {              \
+    static const char* kName;                 \
+    template <typename... Args>               \
+    rocblas_status operator()(Args... args) { \
+      return ::__name(args...);               \
+    }                                         \
+  } __name;                                   \
+  const char* WrapperShim__##__name::kName = #__name;
+
+#else
+
+#define ROCBLAS_API_WRAPPER(__name)                                        \
```

Comment:
That seems to duplicate the macro defined in SE below?

Reply:
it is a duplicate and unfortunately that is by design....see the commit message for the 4th commit in the PR for the details why.

https://github.com/tensorflow/tensorflow/pull/50420/commits/ec71a56e3af33a7c80d41b7fa8112a0ac0d8fc58

Reply:
could we avoid duplication without creating duplicate symbols? maybe define once in stream executor?

Reply:
@cheshire  

I am unable to come up with a way to avoid the duplication :(

The problem is that the functor  (`struct DynLoadShim__##__name `)  instance (with instance name == rocblas api name) needs to be in the header file, so that it can be shared. And because it is in the header file, it runs into the dupilcate declaration issue.

posting linker error here for easy reference
```
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow/stream_executor/rocm/_objs/rocblas_plugin/rocm_blas.o: multiple definition of 'tensorflow::wrap::DynLoadShim__rocblas_create_handle::kName'
/usr/bin/ld.gold: bazel-out/k8-opt/bin/tensorflow/compiler/xla/service/gpu/_objs/cusolver_context/cusolver_context.o: previous definition here
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow/stream_executor/rocm/_objs/rocblas_plugin/rocm_blas.o: multiple definition of 'tensorflow::wrap::DynLoadShim__rocblas_destroy_handle::kName'
/usr/bin/ld.gold: bazel-out/k8-opt/bin/tensorflow/compiler/xla/service/gpu/_objs/cusolver_context/cusolver_context.o: previous definition here
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow/stream_executor/rocm/_objs/rocblas_plugin/rocm_blas.o: multiple definition of 'tensorflow::wrap::DynLoadShim__rocblas_set_stream::kName'
...
...
/usr/bin/ld.gold: bazel-out/k8-opt/bin/tensorflow/compiler/xla/service/gpu/_objs/cusolver_context/cusolver_context.o: previous definition here
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow/stream_executor/rocm/_objs/rocblas_plugin/rocm_blas.o: multiple definition of 'tensorflow::wrap::rocblas_snrm2'
/usr/bin/ld.gold: bazel-out/k8-opt/bin/tensorflow/compiler/xla/service/gpu/_objs/cusolver_context/cusolver_context.o: previous definition here
collect2: error: ld returned 1 exit status
...

The surprising thing is that we do not run into this link error when building the python whl package. We only hit it when running the unit tests. Given the nature of the error, I would have expected this error to show up when linking any library that pulls in both the `cusolver_context.o`  and `rocblas_wrapper.o` (which I would expect to happen when building the TF python whl).

Reply:
@cheshire gentle ping